src/Protos/commands.proto

```diff
@@ -0,0 +1,52 @@
+syntax = "proto3";
+
+import "types.proto";
+
+option csharp_namespace = "ProjectOrigin.Registry.Api";
+
+package ProjectOrigin.v1;
+
+service Registry {
+  rpc SubmitCommand (Command) returns (CommandStatus);
+}
+
+enum CommandState {
+  Failed = 0;
+  Succeeded = 1;
+}
+
+enum CommandLineState {
+  Failed = 0;
+  Rollback = 1;
+  Succeeded = 2;
+}
+
+message CommandStatus {
+  bytes Id = 1;
+  CommandState State = 2;
+  string Error = 3;
+  repeated CommandLineStatus Lines = 4;
+}
+
+message CommandLineStatus {
+  CommandLineState State = 1;
+  string Error = 2;
+}
+
+message Command {
+  message Envelope {
```

Comment:
This is an elegant solution for event-based messages, Envelope is simply awesome